ci_scripts/install-tools.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+# Purpose: Installs or upgrades essential development tools.
+# Notes:
+#   - Supports macOS and Linux (best effort) for installations via `brew install` and `go install`.
+#   - Installs tools like kubectl, helm, pre-commit, go, and chloggen.
+#   - Use OVERRIDE_OS_CHECK=true to bypass OS compatibility checks.
+#   - This script is intended to be run via `make install-tools`.
+#   - Prompts the user for approval to install or update each tool if the tool is out of date.
+#
+# Example Usage:
+#    make install-tools [OVERRIDE_OS_CHECK=true]
+
+# Function to install or upgrade a tool
+install_or_upgrade() {
+  local tool=$1
+  local type=$2
+
+  case $type in
+    brew)
+      install_or_upgrade_brew "$tool"
+      ;;
+    go)
+      install_or_upgrade_go "$tool"
+      ;;
+    *)
+      echo "Unsupported tool type: $type"
+      exit 1
+      ;;
+  esac
+}
+
+# Function to install or upgrade a Homebrew-based tool
+install_or_upgrade_brew() {
+  local tool=$1
+  local installed_version=$(brew list $tool --versions | awk '{print $2}')
+  local latest_version=$(brew info --json=v1 "$tool" | jq -r '.[0].versions.stable')
+
+  if [ "$installed_version" == "$latest_version" ]; then
+    echo "$tool is already up to date (version $installed_version)."
+    return
+  fi
+
+  read -p "$tool version $installed_version is installed. Would you like to upgrade to $latest_version? (y/n): " yn
+  case $yn in
+    [Yy]* )
+      brew upgrade $tool || echo "Failed to upgrade $tool. Continuing..."
+      ;;
+    [Nn]* )
+      echo "Skipping upgrade for $tool."
+      ;;
+  esac
+}
+
+# Function to install or upgrade a Go-based tool
+install_or_upgrade_go() {
+  local tool=$1
+  local tool_path="$LOCALBIN/$(basename $tool)"
+  local installed_version
+
+  if [ -f "$tool_path" ]; then
+    installed_version=$($tool --version 2>/dev/null)  # Try to get the version
+    if [ -z "$installed_version" ]; then  # If version is empty
+      # Fallback to file modification time
+      installed_version="UNKNOWN (Last updated: $(stat -c %y "$tool_path" 2>/dev/null || stat -f "%Sm" "$tool_path"))"
+    fi
+  else
+    installed_version="Not Installed"
+  fi
+
+  read -p "$tool version $installed_version is installed. Would you like to upgrade? (y/n): " yn
```

Comment:
`install_tools` is interactive? Is it necessary?

Reply:
The version control for tools in this implementation, particularly helm and chloggen, are not as robust as in other projects like [opentelemetry-collector-contrib](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/7db0833d735851da11277677e0a955d95055bf83/Makefile.Common#L40). Our current implementation takes the path of least resistance and might encounter version compatibility issues for developers. In the long run, we should consider containerizing these tools or enhancing the "make install-tools" for greater portability.

Reply:
Removed human input. Updated logging to reflect current state and actions executed. Done.